OpenRA.Game/Renderer.cs

```diff
@@ -123,26 +128,49 @@ public void InitializeDepthBuffer(MapGrid mapGrid)
 		public void BeginFrame(int2 scroll, float zoom)
 		{
 			Context.Clear();
+
+			// Cache property results to avoid the unnecessary overhead of repeat queries
+			var resolution = Resolution;
+			var scale = WindowScale;
+
+			var backingResolution = new Size((int)(scale * resolution.Width), (int)(scale * resolution.Height));
```

Comment:
Expose `IPlatformWindow.SurfaceSize` directly instead of estimating it via floating point math.